### DISCUSSION TITLE
More thread with `duckdb_execute_prepared_streaming`

### DISCUSSION THREAD

**Discussion by OkobaPatino** [duckdb/duckdb#13448] [Q&A]
When I use `duckdb_execute_prepared_streaming` instead of `duckdb_execute_prepared`, it works only with one thread and slows down significantly.
I do not want to use `duckdb_execute_prepared` as it loads all the file into memory and I wish to have a lower memory usage using `duckdb_execute_prepared_streaming` while keeping the speed.
Is that possible?

**Comment by Tishj** [comment]
<https://github.com/duckdb/duckdb/pull/11494> should fix this in v1.1.0 / nightly

**Reply by OkobaPatino** [reply]
I go the nightly version but how should I use this new feature? Beside calling these: 
`duckdb_prepare`
`duckdb_execute_prepared_streaming`
`duckdb_stream_fetch_chunk`

**Reply by Tishj** [reply]
It's just internals, you don't need to make any changes to make use of it

The problem before was that if the source supported batch indexing we had to fall back to single threaded execution for that pipeline, because the sink (the result collector) didn't support batch indexing in parallel

**Reply by OkobaPatino** [reply]
So it depends on the source and for example for `CSV` it still falls back to single thread.

**Reply by OkobaPatino** [reply]
As I want to use the `CSV` option, will it be supported or am I missing something?
